test_proxy_auth.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python3
```

Comment:
Let's move all the tests in a directory (same for all the other PRs) and please add instructions on how to run the tests.
I think `python -m unittest` should work (maybe with some arguments to point the test folder)?